prow/plugins/dog/dog.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package dog adds dog images to issues in response to a /woof comment
+package dog
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+var (
+	match = regexp.MustCompile(`(?mi)^/woof\s*$`)
+)
+
+const (
+	dogURL     = realClowder("https://random.dog/woof.json")
+	pluginName = "dog"
+)
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericComment, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+	// The Config field is omitted because this plugin is not configurable.
+	// TODO(qhuynh96): Removes all the fields of pluginHelp except Description.
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The dog plugin adds a dog image to an issue in response to the `/woof` command.",
+		WhoCanUse:   "Anyone",
+		Usage:       "/woof",
+		Examples:    []string{"/woof"},
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/woof",
+		Description: "Add a dog image to the issue",
+		Featured:    false,
+		WhoCanUse:   "Anyone",
+		Examples:    []string{"/woof"},
+	})
+	return pluginHelp, nil
+}
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+}
+
+type clowder interface {
```

Comment:
A pack of dogs, not a clowder

Reply:
done, somehow I thought this was some meme talk related to using cloud APIs :shrug: